### ISSUE TITLE
`Layout/RescueEnsureAlignment` does not properly register offenses when the send node of a block is split over multiple lines

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#14529]
## Is your feature request related to a problem? Please describe.

With the following `.rubocop.yml`:
```yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4

Layout/BlockAlignment:
  EnforcedStyleAlignWith: 'start_of_block'
```

And the following file `foo.rb`:
```ruby
# frozen_string_literal: true

Foo
  .map do |x|
    x + 1
rescue StandardError
  puts 'error!'
  end
```

When you run `rubocop foo.rb`, no offenses are detected.

## Describe the solution you'd like

I would like if an offense was detected, and the above code can be autocorrected to:

```ruby
# frozen_string_literal: true

Foo
  .map do |x|
    x + 1
  rescue StandardError
    puts 'error!'
  end
```

## Additional context

I think this is in the spirit of the `EnforcedStyleAlignWith` configuration, but its kinda a weird case since it kinda also relates to `Layout/RescueEnsureAlignment` cop, so I'm not sure which cop this should be related to.

I wasn't exactly sure how to categorize this issue, so I chose feature request, but feel free to change the label.

**Comment by dvandersluis** [comment]
What version of RuboCop are you using? This works as expected for me on 1.80.2:

```
Inspecting 1 file
C

Offenses:

test.rb:5:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
  x + 1

test.rb:8:1: C: [Corrected] Layout/BlockAlignment: end at 8, 0 is not aligned with .map do |x| at 4, 2.
end
^^^
```

`test.rb` is autocorrected to:

```ruby
# frozen_string_literal: true

Foo
  .map do |x|
    x + 1
  rescue StandardError
    puts 'error!'
  end
```

**Comment by a-lavis** [comment]
> What version of RuboCop are you using? This works as expected for me on 1.80.2:

`rubocop -V`
```
1.80.2 (using Parser 3.3.9.0, Prism 1.4.0, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.4) [arm64-darwin24]
```

Huh, weird.  I triple checked and I'm not getting any offenses on the file I posted above
> ```ruby
> # frozen_string_literal: true
> 
> Foo
>   .map do |x|
>     x + 1
> rescue StandardError
>   puts 'error!'
>   end
> ```

**Comment by a-lavis** [comment]
`cat foo.rb`
```
# frozen_string_literal: true

Foo
  .map do |x|
    x + 1
rescue StandardError
  puts 'error!'
  end
```
`rubocop -d foo.rb`
```
For /Users/aidanlavis/test-rubocop: configuration from /Users/aidanlavis/test-rubocop/.rubocop.yml
Default configuration from /Users/aidanlavis/.rbenv/versions/3.4.4/lib/ruby/gems/3.4.0/gems/rubocop-1.80.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/aidanlavis/test-rubocop/foo.rb
Loading cache from /Users/aidanlavis/.cache/rubocop_cache/4e4d008a50543e03c4c8ef98d2aa6e56c5c981e3/6d7a3b621ca1730e04accd938619e4bdab66cfb1/c77a0a24d88b94df732d4ac966bb355e17d61059
.

1 file inspected, no offenses detected
Finished in 0.046703000087291 seconds
```

**Comment by dvandersluis** [comment]
Can you create a config file `test.yml` with the following, and run `rubocop -c test.yml foo.rb -A`:

```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4

Layout/BlockAlignment:
  EnforcedStyleAlignWith: start_of_block
```

If this works, please provide a .rubocop.yml configuration that causes the offense to not be registered.

**Comment by a-lavis** [comment]
`cat test.yml`
```
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4

Layout/BlockAlignment:
  EnforcedStyleAlignWith: start_of_block
```

`cat foo.rb`
```
# frozen_string_literal: true

Foo
  .map do |x|
    x + 1
rescue StandardError
  puts 'error!'
  end
```

`rubocop -c test.yml foo.rb -A`
```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

`rubocop -d -c test.yml foo.rb -A`
```
configuration from /Users/aidanlavis/test-rubocop/test.yml
Default configuration from /Users/aidanlavis/.rbenv/versions/3.4.4/lib/ruby/gems/3.4.0/gems/rubocop-1.80.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/aidanlavis/test-rubocop/foo.rb
Loading cache from /Users/aidanlavis/.cache/rubocop_cache/4e4d008a50543e03c4c8ef98d2aa6e56c5c981e3/1d033f73fc0fff7b85903592a7b47c5de98d00f3/c77a0a24d88b94df732d4ac966bb355e17d61059
.

1 file inspected, no offenses detected
Finished in 0.047110999934375286 seconds
```

`rubocop -V`
```
1.80.2 (using Parser 3.3.9.0, Prism 1.4.0, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.4) [arm64-darwin24]
```
still not registering any offenses 🤔

**Comment by dvandersluis** [comment]
Can you try with `-C false` (disable the cache)?

**Comment by a-lavis** [comment]
`rubocop -d -C false -c test.yml foo.rb -A`
```
configuration from /Users/aidanlavis/test-rubocop/test.yml
Default configuration from /Users/aidanlavis/.rbenv/versions/3.4.4/lib/ruby/gems/3.4.0/gems/rubocop-1.80.2/config/default.yml
Inspecting 1 file
Scanning /Users/aidanlavis/test-rubocop/foo.rb
.

1 file inspected, no offenses detected
Finished in 0.06956800003536046 seconds
```

**Comment by a-lavis** [comment]
in case it could be helpful:

`which rubocop`
```
/Users/aidanlavis/.rbenv/shims/rubocop
```
`rbenv which rubocop`
```
/Users/aidanlavis/.rbenv/versions/3.4.4/bin/rubocop
```

**Comment by dvandersluis** [comment]
Thank you, it seems my IDE reformatted your example and unindented the final `end`. I can reproduce now. Sorry for the confusion!

**Comment by dvandersluis** [comment]
It looks like this should actually be registered as an offense by `Layout/RescueEnsureAlignment` but is a bug (the `end` is actually properly aligned to `start_of_block`!). I'm looking into it.
